specs/refactor_common_code_and_move.adoc

```diff
@@ -0,0 +1,206 @@
+= Refactor modules like tendrl-gluster-integartion, tendrl-node-agent etc and
+move the common code to tendrl-common
+
+
+All the Tendrl modules like tendrl-gluster-integartion, tendrl-node-agent etc
+use various persisters, managers and job executors which have almost similar
+logic written and could ideally be moved to the common module tendrl-common.
+
+Any module specific logic can be implemented as part of extended classes from
+these common ones defined in tendrl-common module.
+
+
+== Problem description
+
+Currently there are lot of duplicate classes, code written across Tendrl modules
+like persister for central store, manager, job handlers from the etcd queue.
+Also base classes for flows, atoms and test cases are replicated across modules.
+
+These replicated classes and sources could be moved to the common module and
+individual modules can either use them as is or extend them for additional
+business logic.
+
+Even there are various sample configuration files which are replicated across
+the tendrl modules and could be more organized and moved to common module. Only
+modules specific configurations should be kept as part of the individual modules
+
+
+== Use Cases
+
+None
```

Comment:
The most important use case is that this enables commons to function as a library and the code becomes completely reusable by any component.